### PR TITLE
PR for adding distillation training

### PR DESCRIPTION
**Pull request by thakur-nandan**
@MXueguang @crystina-z here is the PR to fine-tune a retriever model using Distillation on the latest v2 branch.
This code requires the scores to be prepopulated using the reranker model under the field `score` in the training dataset.

The majority of the code is copied from the contrastive loss function.

### CODE REVIEW

**Comment by MXueguang** [comment]
lol seems copilot is pretty good at catch typo
